src/lib/customHooks.ts

```diff
@@ -0,0 +1,33 @@
+import { usersApi } from '@/api/api';
+import { useState, useEffect } from 'react';
+import { toUser, User } from './user';
+
+export function useUser() {
+  const [user, setUser] = useState<User | null>(null);
+
+  const getAuthenticatedUser = (): Promise<User | null> => {
+    return new Promise((resolve, reject) => {
+      usersApi
+        .usersControllerGetProfile()
+        .then((response) => {
+          resolve(toUser(response));
+        })
+        .catch(() => {
+          reject(null);
+        });
```

Comment:
`resolve` and `reject` not needed, it's already a Promise

Reply:
I am creating a new Promise here by taking an autogenerated `usersControllerGetProfile` function and modifying its result with the help of `toUser` method to be either a nice clean `User` object or null. And since I cannot make this function `async` I have to return a new Promise instead of `toUser(await usersApi.usersControllerGetProfile())`. I don't really see a way to make it work otherwise.

Reply:
but inside `then` the promise is resolved, you don't need to create another Promise

Reply:
Not sure if I understand what you mean. Maybe you could give me a code example? In my function `then` resolves the Promise returned by `usersControllerGetProfile()` and gives me the raw response data and then I resolve the new Promise returning this response as a `User` or null.